app/src/org/commcare/preferences/HiddenPreferences.java

```diff
@@ -446,4 +448,26 @@ public static void enableBypassPreUpdateSync(boolean enable) {
     public static boolean shouldBypassPreUpdateSync() {
         return CommCareApplication.instance().getCurrentApp().getAppPreferences().getBoolean(BYPASS_PRE_UPDATE_SYNC, false);
     }
+
+    public static void skipCommCareVersionForUpdate(int version) {
+        CommCareApplication.instance().getCurrentApp().getAppPreferences()
+                .edit()
+                .putInt(SKIP_COMMCARE_VERSION_FOR_UPDATE, version)
+                .apply();
+    }
+
+    public static int getSkippedCommCareUpdateVersion() {
+        return CommCareApplication.instance().getCurrentApp().getAppPreferences().getInt(SKIP_COMMCARE_VERSION_FOR_UPDATE, -1);
+    }
+
+    public static void setLastInAppUpdateCheckTime(long millis) {
```

Comment:
this should be saved in Global prefs rather than app prefs as it's going to be same across all CommCare apps installed on CommCare.